arangod/Graph/GraphManager.cpp

```diff
@@ -550,7 +555,18 @@ Result GraphManager::ensureCollections(Graph* graph, bool waitForSync) const {
       return res;
     } else {
       if (edgeCollectionsToCreate.find(vertexColl) == edgeCollectionsToCreate.end()) {
-        documentCollectionsToCreate.emplace(vertexColl);
+        if (graph->isHybrid()) {
+          if (satelliteDocumentCollectionsToCreate.find(vertexColl) ==
+                  satelliteDocumentCollectionsToCreate.end() &&
+              graph->satelliteCollections().find(vertexColl) !=
```

Comment:
Marge this if with the one above:
if hybrid() && collection found in satellites
this way you can eliminate one of the false cases below.

Reply:
thanks! done